src/core/public/application/types.ts

```diff
@@ -44,6 +45,11 @@ export interface AppBase {
    */
   title: string;
 
+  /**
+   * The category the app lives in
+   */
+  category?: AppCategory;
```

Comment:
Related to @pgayvallet's comments [here](https://github.com/elastic/kibana/pull/53545#discussion_r365596669) and [here](https://github.com/elastic/kibana/pull/53545#discussion_r365597177).

Instead of this property being an enum value, it seems like this would be more flexible if it was instead an object with a label and a iconType value:

```ts
export interface AppCategory {
  label: string;
  ariaLabel: string;
  euiIconType?: string;
  order?: number; // similar to App.order
}

export interface AppBase {
  category?: AppCategory;
}
```

For the built-in categories, we could statically export a constant with the appropriate values:

```ts
export const APP_CATEGORIES = Object.freeze({
  management: {
    label: i18n.translate(...),
    ariaLabel: i18n.translate(...),
    euiIconType: '...',
    order: 1000,
  },
  analyze: {
    ...
  },

  ...
}
```

This would be used like:
```ts
import { APP_CATEGORIES } from 'src/core/public';

class Plugin {
  setup(core) {
    core.application.register({
      id: 'blah',
      category: APP_CATEGORIES.analyze,
      ...
    });
  }
}
```

But 3rd party plugins could easily provide their own custom categories:
```ts
class Plugin {
  setup(core) {
    core.application.register({
      id: 'blah',
      category: {
        label: 'My Company',
      },
      ...
    });
  }
}
```

To me this seems like an easy way to get support for custom categories without making using the built-in categories any more work.

With this design, you can also remove the `getGroupLabel` and `getGroupIcon` functions from the Header code.